### PR TITLE
Feature: Exclude file types

### LINKED ISSUE #13: Add command line argument to ignore file types
## Description

The current implementation of the config-file-validator will scan and report on all supported file types that it discovers in the search path. This may not be desired behavior for some teams that just want to target a subset of the supported configuration file types. We should add a CLI argument to allow users to specify a list of supported file types to exclude from the validation run. 

## Potential Solution
```
validator --search-path=/path/to/search --exclude-file-types=yaml,json
```

### PR DESCRIPTION
**Pull request by ravsii**
Resolves #13

### CODE REVIEW

**Comment by ravsii** [comment]
Hey @kehoecj, take a look. I closed the old one #16 because it was easier to just start from scratch.

I saw you took some of my ideas with option funcs and stuff, appreciate it 👍

**Comment by kehoecj** [comment]
> Hey @kehoecj, take a look. I closed the old one #16 because it was easier to just start from scratch.
> 
> 
> 
> I saw you took some of my ideas with option funcs and stuff, appreciate it 👍 

Thank you for implementing this, you had some great suggestions! We'll take a look and get this merged

**Review by jackswiney — APPROVED** [review]
LGTM

File: cmd/validator/validator.go [hunk 1]
```diff
@@ -10,9 +10,11 @@ Usage:
 
 The flags are:
     -exclude-dirs string
-    	Subdirectories to exclude when searching for configuration files
+    	Subdirectories to exclude when searching for configuration files.
 	-reporter string
 		Format of printed report. Currently supports standard, JSON.
+	-exclude-file-types string
```
Comment (jd4235) [inline cmd/validator/validator.go:16]: Looks like these comments got misaligned at some point with a tab starting at line 14. Please correct to replace tabs with spaces and align the options to line 12.

File: pkg/finder/fsfinder.go [hunk 2]
```diff
@@ -102,3 +114,15 @@ func (fsf FileSystemFinder) Find() ([]FileMetadata, error) {
 
 	return matchingFiles, nil
 }
+
+// isExtensionExcluded returns true extension exists in exclude list.
```
Comment (jd4235) [inline pkg/finder/fsfinder.go:118]:
```suggestion
// isExtensionExcluded returns true if extension exists in exclude list.
```

File: pkg/finder/finder_test.go [hunk 3]
```diff
@@ -40,6 +42,24 @@ func Test_fsFinderExcludeDirs(t *testing.T) {
 	}
 }
 
+func Test_fsFinderExcludeFileTypes(t *testing.T) {
```
Comment (jd4235) [inline pkg/finder/finder_test.go:45]: I see you added this test but I don't see it ever executed in the test stage of the pipeline.  Please add a retest.

**Review by jd4235 — CHANGES_REQUESTED** [review]
See comments. Will re-review once resolved.

File: pkg/finder/fsfinder.go [hunk 4]
```diff
@@ -102,3 +114,15 @@ func (fsf FileSystemFinder) Find() ([]FileMetadata, error) {
...
+// TODO: refactor to slices.Contains when project will be updated to go 1.21.
```
[the remark this answers is not in the document]
Comment (kehoecj) [reply pkg/finder/fsfinder.go:119]: Nope, I'll go ahead and create the issue and the code can get updated to remove the TODO with a link to the issue

[the remark this answers is not in the document]
Comment (kehoecj) [reply pkg/finder/fsfinder.go:119]: https://github.com/Boeing/config-file-validator/issues/36

[the remark this answers is not in the document]
Comment (ravsii) [reply pkg/finder/fsfinder.go:119]: Or we can just update it go 1.21, but that's for you guys to decide, maybe you need to keep it 1.18 for your other projects.
afaik there are no breaking changes between these versions

**Comment by ravsii** [comment]
Sure, I'll fix it as soon as I can

File: pkg/finder/fsfinder.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (kehoecj) [reply pkg/finder/fsfinder.go:119]: I think I'd like to tackle moving to 1.21 in a separate story/MR just to keep this one in scope of the existing change

File: pkg/finder/finder_test.go (hunk 3 above)
Comment (ravsii) [reply pkg/finder/finder_test.go:45]: https://github.com/Boeing/config-file-validator/actions/runs/6189865772/job/16888184708#step:4:76

I added tests in later commits, perhaps you just checked the wrong pipeline

**Comment by jd4235** [comment]
LGTM

**Pull request merged**